addons/mail/static/tests/message/message_tests.js

```diff
@@ -42,6 +43,40 @@ QUnit.test("Start edition on click edit", async () => {
     await contains(".o-mail-Message-editable .o-mail-Composer-input", { value: "Hello world" });
 });
 
+QUnit.test("Editing message keeps the mentioned channels", async () => {
+    const pyEnv = await startServer();
+    const channelId1 = pyEnv["discuss.channel"].create({
+        name: "general",
+        channel_type: "channel",
+    });
+    const channelId2 = pyEnv["discuss.channel"].create({
+        name: "other",
+        channel_type: "channel",
+    });
+    pyEnv["mail.message"].create({
+        author_id: pyEnv.currentPartnerId,
+        body: markup(
+            `<span><a href='http://test.odoo.com/web#model='discuss.channel'&amp;id=${channelId2}' class='o_channel_redirect' data-oe-id='${channelId2}' data-oe-model='discuss.channel' target='_blank'>#other</a> hello it's me</span>`
+        ),
+        model: "discuss.channel",
+        res_id: channelId1,
+        message_type: "comment",
+    });
```

Comment:
Instead of forging the message with mention by hand, would be best to post a new message with channel mention.

The likelihood that the format of channel mention will change drastically, especially with markdown and the intent to have live preview in composer eventually (which requires a markdown-compliant format on top of it for making a reliable tiny WYSIWIG)